### PR TITLE
feat(utils/function): clamp function add

### LINKED ISSUE #57: Create clamp Utility Function
### `clamp`

Clamps a number within the inclusive range specified by the min and max values.

**File**: `src/utils/math/clamp/clamp.ts`
**Test File**: `src/utils/math/clamp/clamp.test.ts`

**Parameters**:
- `num: number`: The number to clamp.
- `min: number`: The minimum value.
- `max: number`: The maximum value.

**Returns**:
- `number`: The clamped number.

**Example**:
```ts
clamp(5, 1, 10);    // 5
clamp(-3, 0, 10);   // 0
clamp(15, 1, 10);   // 10
```

### PR DESCRIPTION
**Pull request by vigneshbirru**
Closes #57 
This PR #57  adds input validation to the clamp function, ensuring all parameters are numbers and the minimum value is not greater than the maximum. It improves error messaging for invalid inputs.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `clamp` function to ensure values stay within a specified range, enhancing data validation and integrity.
	- Added a comprehensive test suite for the `clamp` function to validate its behavior across various scenarios, including edge cases and error handling.
	- Integrated the `clamp` function into the utility index for broader accessibility.

### CODE REVIEW

**Comment by bharat-dussa** [comment]
PR title is not correct

File: src/utils/clamp/clamp.ts
Comment (bharat-dussa) [inline src/utils/clamp/clamp.ts:1]: path should be in `utils/function/clamp/clamp.ts`

File: clamp.js
Comment (bharat-dussa) [inline clamp.js:1]: why is this file exists here ?

File: src/utils/clamp/clamp.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export function clamp(num: number, min: number, max: number): number {
+    if (typeof num !== 'number' || typeof min !== 'number' || typeof max !== 'number') {
+        throw new TypeError('All arguments must be numbers');
```
Comment (bharat-dussa) [inline src/utils/clamp/clamp.ts:3]:
```suggestion
        throw new TypeError('Invalid input: all parameters must be numbers.');
```

File: src/utils/clamp/clamp.test.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        expect(clamp(15, 1, 10)).toBe(10);
+    });
+
+    test('should throw a TypeError if non-number arguments are passed', () => {
+        expect(() => clamp('5' as any, 1, 10)).toThrow(TypeError);
+    });
+
+    test('should throw a RangeError if min is greater than max', () => {
+        expect(() => clamp(5, 10, 1)).toThrow(RangeError);
+    });
```
Comment (bharat-dussa) [inline src/utils/clamp/clamp.test.ts:23]: add test for max number 
clamp(1e20, 1e10, 1e15)

File: src/utils/clamp/clamp.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export function clamp(num: number, min: number, max: number): number {
```
Comment (bharat-dussa) [inline src/utils/clamp/clamp.ts:1]: add jsDoc for this

/**
 * Clamps a value between a minimum and maximum range.
 * 
 * @param value - The value to be clamped.
 * @param min - The minimum value.
 * @param max - The maximum value.
 * @returns The clamped value.
 */
```suggestion
export function clamp(value: number, min: number, max: number): number {
```

File: src/utils/clamp/clamp.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    if (typeof num !== 'number' || typeof min !== 'number' || typeof max !== 'number') {
```
Comment (bharat-dussa) [inline src/utils/clamp/clamp.ts:2]:
```suggestion
    if (typeof value !== 'number' || typeof min !== 'number' || typeof max !== 'number') {
```

File: src/utils/clamp/clamp.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+        throw new TypeError('All arguments must be numbers');
+    }
+
+    if (min > max) {
+        throw new RangeError('Minimum value cannot be greater than maximum value');
+    }
+
+    return Math.max(min, Math.min(num, max));
```
Comment (bharat-dussa) [inline src/utils/clamp/clamp.ts:10]:
```suggestion
    return Math.max(min, Math.min(value, max));
```

**Review by bharat-dussa — CHANGES_REQUESTED** [review]

File: yarn.lock
Comment (bharat-dussa) [inline yarn.lock:1]: don't push yarn.lock

**Review by bharat-dussa — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (bharat-dussa) [inline package-lock.json:1]: Why this File is here ?

File: clamp.js
Comment (bharat-dussa) [inline clamp.js:1]: Why this file is here ?

File: yarn.lock
Comment (bharat-dussa) [inline yarn.lock:1]: Why this file is here?

**Review by bharat-dussa — CHANGES_REQUESTED** [review]
